### PR TITLE
Fix part of #2637: Added accessibilty label to the RevisionCardActivity

### PR DESCRIPTION
**Pull request by jonathanalvares9009**
Whenever RevisionCardActivity was clicked it used to say Oppia instead changed it to Skill page.
Fixes part of #2637
## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Review by MaskedCarrot — APPROVED** [review]
LGTM,thanks.

**Comment by MaskedCarrot** [comment]
@jonathanalvares9009 the test in RevisionCardActivityTest are failing can you please check. Also can you please upload a screenshot showing the tests passing for you?

**Comment by MaskedCarrot** [comment]
There are no dagger modules in `RevisionCardActivityTest`
You can also take a look at the test file of this pr https://github.com/oppia/oppia-android/pull/2964/files to learn more.

**Comment by jonathanalvares9009** [comment]
@MaskedCarrot I had done the changes can you check?

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+  interface TestApplicationComponent : ApplicationComponent {
+    @Component.Builder
+    interface Builder : ApplicationComponent.Builder
+
+    fun inject(revisionCardActivityTest: RevisionCardActivityTest)
+  }
+
+  class TestApplication : Application(), ActivityComponentFactory, ApplicationInjectorProvider {
+    private val component: RevisionCardActivityTest.TestApplicationComponent by lazy {
```
Comment (MaskedCarrot) [inline app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: remove the redundant qualifier `RevisionCardActivityTest`

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+      DaggerRevisionCardActivityTest_TestApplicationComponent.builder()
+        .setApplication(this)
+        .build() as RevisionCardActivityTest.TestApplicationComponent
```
Comment (MaskedCarrot) [inline app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: remove the redundant qualifier `RevisionCardActivityTest`

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    topicId: String,
+    subTopicId: Int
+  ): Intent {
+    return RevisionCardActivity.createRevisionCardActivityIntent(
+      ApplicationProvider.getApplicationContext(),
+      internalProfileId,
+      topicId,
+      subTopicId
+    )
+  }
```
Comment (MaskedCarrot) [inline app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: move the private function after the tests and before the dagger modules.

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -436,4 +436,5 @@
   <string name="coming_soon">Coming Soon</string>
   <string name="recommended_stories">Recommended Stories</string>
   <string name="stories_for_you">Stories For You</string>
+  <string name="revision_card_activity_title">Skill page</string>
```
Comment (MaskedCarrot) [inline app/src/main/res/values/strings.xml]: @rt4914 should this come under the comment `<!-- RevisionCardActivity -->`

**Review by MaskedCarrot — CHANGES_REQUESTED** [review]
@jonathanalvares9009 PTAL at the following.

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt (hunk 3 above)
Comment (MaskedCarrot) [reply app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: same for the other private function.

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+  @get:Rule
+  val activityTestRule: ActivityTestRule<RevisionCardActivity> = ActivityTestRule(
+    RevisionCardActivity::class.java,
+    /* initialTouchMode= */ true,
+    /* launchActivity= */ false
+  )
+
+  @Test
+  fun testRevisionCardActivity_hasCorrectLabel() {
```
Comment (MaskedCarrot) [inline app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: change the function name to `testRevisionCardActivity_hasCorrectActivityLabel`

**Comment by MaskedCarrot** [comment]
@jonathanalvares9009 it is a good practice to write done on the suggested changes once you apply them.

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt (hunk 1 above)
Comment (jonathanalvares9009) [reply app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: Done

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt (hunk 2 above)
Comment (jonathanalvares9009) [reply app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: Done

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt (hunk 3 above)
Comment (jonathanalvares9009) [reply app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: Done

**Comment by anandwana001** [comment]
Trying re-running job. 
Let's wait for the reviewer to review.

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt (hunk 5 above)
Comment (jonathanalvares9009) [reply app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: Done

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (jonathanalvares9009) [reply app/src/main/res/values/strings.xml]: Didn't do this because not string label has a comment and I thought it would be out of the style that is being followed

**Comment by jonathanalvares9009** [comment]
@rt4914 can you review this PR?

**Comment by jonathanalvares9009** [comment]
@MaskedCarrot can you review my PR?

**Review by MaskedCarrot — APPROVED** [review]
LGTM, Thanks

**Comment by MaskedCarrot** [comment]
@jonathanalvares9009 the PR  can only be merged after it is approved by the other reviewers, This can take a day or two after the weekend. In the meantime, you can try working on some other issues.

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (rt4914) [reply app/src/main/res/values/strings.xml]: @MaskedCarrot Its fine. Currently in `strings.xml` file we are not focussed on comments and order of strings.

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    /* initialTouchMode= */ true,
+    /* launchActivity= */ false
+  )
+
+  @Inject
+  lateinit var context: Context
+
+  private val internalProfileId = 0
+
+  private val subTopicId = 0
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: Use `1`.
`private val subTopicId = 1`

Subtopic Ids start from `1`.

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testRevisionCardActivity_hasCorrectActivityLabel() {
+    activityTestRule.launchActivity(
+      createRevisionCardActivityIntent(
+        internalProfileId = internalProfileId,
+        topicId = TEST_TOPIC_ID_0,
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: Use `FRACTIONS_TOPIC_ID` and once that is done remove any unused imports.

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+    )
+    val title = activityTestRule.activity.title
+
+    // Verify that the activity label is correct as a proxy to verify TalkBack will announce the
+    // correct string when it's read out.
+    assertThat(title).isEqualTo(context.getString(R.string.revision_card_activity_title))
+  }
+
+  private fun setUpTestApplicationComponent() {
+    ApplicationProvider.getApplicationContext<RevisionCardActivityTest.TestApplication>()
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: Nit fix for this line
`ApplicationProvider.getApplicationContext<TestApplication>().inject(this)`

**Review by rt4914 — CHANGES_REQUESTED** [review]
@jonathanalvares9009 I have suggested some change PTAL.

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt (hunk 6 above)
Comment (jonathanalvares9009) [reply app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: Made the change @rt4914

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt (hunk 7 above)
Comment (jonathanalvares9009) [reply app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: Used FRACTIONS_TOPIC_ID instead of TEST_TOPIC_ID_0

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt (hunk 8 above)
Comment (jonathanalvares9009) [reply app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: Done

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (BenHenning) [inline app/src/main/res/values/strings.xml]: @rt4914 why 'skill page'? That doesn't seem very informational given the context. Maybe 'skill revision page' would be clearer?

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+    /* initialTouchMode= */ true,
+    /* launchActivity= */ false
+  )
+
+  @Inject
+  lateinit var context: Context
+
+  private val internalProfileId = 0
+
+  private val subTopicId = 1
```
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: Nit: ``subtopicId`` ('subtopic' is one word).

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+    return RevisionCardActivity.createRevisionCardActivityIntent(
+      ApplicationProvider.getApplicationContext(),
+      internalProfileId,
+      topicId,
+      subTopicId
+    )
+  }
+
+  // TODO(#59): Figure out a way to reuse modules instead of needing to re-declare them.
+  // TODO(#1675): Add NetworkModule once data module is migrated off of Moshi.
```
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: Please omit this since it's not actually needed (per #3025).

**Review by BenHenning — COMMENTED** [review]
Thanks @jonathanalvares9009! Just had a few comments, plus one for Rajat. PTAL.

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt (hunk 10 above)
Comment (jonathanalvares9009) [reply app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: Done

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt (hunk 9 above)
Comment (jonathanalvares9009) [reply app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: Done

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (jonathanalvares9009) [reply app/src/main/res/values/strings.xml]: @rt4914 shall I rephrase it to skill revision page

**Comment by MaskedCarrot** [comment]
@jonathanalvares9009 suggested changes should only be marked as resolved by the reviewers.

**Comment by jonathanalvares9009** [comment]
Sorry @MaskedCarrot will remember it next time

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (rt4914) [reply app/src/main/res/values/strings.xml]: I agree that `Skill revision page` would give more context as the previous page/tab is related to revision.

Comment (jonathanalvares9009) [reply app/src/main/res/values/strings.xml]: Done

**Comment by jonathanalvares9009** [comment]
@rt4914 and @BenHenning can you review? I made the changes

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt [hunk 11]
```diff
@@ -109,18 +109,16 @@ class RevisionCardActivityTest {
...
     return RevisionCardActivity.createRevisionCardActivityIntent(
       ApplicationProvider.getApplicationContext(),
       internalProfileId,
       topicId,
-      subTopicId
+      subtopicId
     )
   }
 
-  // TODO(#59): Figure out a way to reuse modules instead of needing to re-declare them.
```
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: This one should be kept; my earlier comment was only referring to the second of the TODOs.

**Review by BenHenning — COMMENTED** [review]
Thanks @jonathanalvares9009 & @rt4914.

Just had one more comment @jonathanalvares9009, PTAL.

File: app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt (hunk 11 above)
Comment (jonathanalvares9009) [reply app/src/sharedTest/java/org/oppia/android/app/topic/revisioncard/RevisionCardActivityTest.kt]: Sorry, @BenHenning misunderstood. Corrected it now.

**Comment by jonathanalvares9009** [comment]
@rt4914 can you assign @BenHenning

**Review by BenHenning — APPROVED** [review]
Thanks @jonathanalvares9009! LGTM.
